docs/adr/000x-jobs-and-notifications.md

```diff
@@ -0,0 +1,110 @@
+   - Notify provide a [Ruby gem][notify-gem] that makes it easy to integrate the service into our app, without specific knowledge of the underlying APIs
+   - We can use the [Notify dashboard][notify-features] to see emails we have sent, logs for individual emails, as well as aggregate stats over all emails
+   - Notify provides the facility to restrict the recipients of emails to a [team or whitelist][notify-whitelist], which we can use to restrict notifications outside of production.
+
+In order to integrate Notify with our app, we will setup 3 'GOV.UK Publishing' accounts as follows:
+
+   - One account will be for production use by any GOV.UK publishing app
+   - The other two accounts will be for testing notifications on integration and staging
+
+The generic 'GOV.UK Publishing' name is to encourage sharing of infrastructure by future publishing apps.
+
+### Action Mailer
+
+We will use [Action Mailer][action-mailer] to generate the individual emails to send to notify:
+
+   - Using Action Mailer allows us to switch to a different delivery mechanism in future
+   - Using Action Mailer to process email is the recommended approach for Rails apps
+
+Although Notify provides a [templating feature][notify-features], we will use a single, generic template for all 'GOV.UK Publishing' notifications, with recipient, subject and body parameters; the body supports a limited form of Markdown. Using Action Mailer to handle the generation of emails means we limit our reliance on Notify templates; we can make most of our changes in code, with the usual benefits of testing, faster debugging and version control.
+
+Notify only supports a [single recipient for each email][notify-gem-send-email], so we will need to generate and send separate emails for each recipient. Although this requires more API calls, it's easier to reason about failure for individual emails.
+
+Notify will reject an email when a whitelist is enabled and the recipient is not in the list. In order to retain end-to-end email functionality but avoid emailing users outside of production, we will use an [Action Mailer Interceptor][action-mailer-interceptor] to redirect emails to a Google Group in integration and staging. The Google Group(s) will act as a dumping ground for viewing the emails that have been generated, which enables us to check they appear correctly in text and HTML forms.
+
+### Active Job
+
+We will use [Active Job][active-job] in combination with [Action Mailer][action-mailer] to send emails asynchronously.
+
+   - This means any delay in calling the Notify APIs does not impact on the user request
+   - We can use the [retry feature][active-job-retry] of Active Job to cope with transient failures from Notify
+
+Using Active Job also allows us to switch to a different queueing provider in future. Initially, we will use [Sidekiq][sidekiq], which is one of the [in-built queueing adapters][active-job-sidekiq] provided with Active Job.
```

Comment:
It'd be worth stating here why we're using Active Job instead of Sidekiq (which I believe is because of the Action Mailer interface and improvements in Active Job retry mechanisms). Since GOV.UK has a convention of Sidekiq over Active Job this is worth justifying more as it's likely a WTF.